homeassistant/components/airzone/binary_sensor.py

```diff
@@ -0,0 +1,103 @@
+"""Support for the Airzone sensors."""
+from __future__ import annotations
+
+from collections.abc import Mapping
+from typing import Any, Final
+
+from aioairzone.const import (
+    AZD_AIR_DEMAND,
+    AZD_DEMAND,
+    AZD_ERRORS,
+    AZD_FLOOR_DEMAND,
+    AZD_NAME,
+    AZD_PROBLEMS,
+    AZD_ZONES,
+)
+
+from homeassistant.components.binary_sensor import (
+    DEVICE_CLASS_PROBLEM,
+    DEVICE_CLASS_RUNNING,
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import AirzoneEntity
+from .const import DOMAIN
+from .coordinator import AirzoneUpdateCoordinator
+
+BINARY_SENSOR_TYPES: Final[tuple[BinarySensorEntityDescription, ...]] = (
+    BinarySensorEntityDescription(
+        device_class=DEVICE_CLASS_RUNNING,
+        key=AZD_DEMAND,
+        name="Demand",
+    ),
+    BinarySensorEntityDescription(
+        device_class=DEVICE_CLASS_PROBLEM,
+        key=AZD_PROBLEMS,
+        name="Problem",
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Add Airzone binary sensors from a config_entry."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+
+    binary_sensors = []
+    for system_zone_id, zone_data in coordinator.data[AZD_ZONES].items():
+        for description in BINARY_SENSOR_TYPES:
+            if description.key in zone_data:
+                binary_sensors.append(
+                    AirzoneBinarySensor(
+                        coordinator,
+                        description,
+                        entry,
+                        system_zone_id,
+                        zone_data,
+                    )
+                )
+
+    async_add_entities(binary_sensors)
+
+
+class AirzoneBinarySensor(AirzoneEntity, BinarySensorEntity):
+    """Define an Airzone sensor."""
+
+    def __init__(
+        self,
+        coordinator: AirzoneUpdateCoordinator,
+        description: BinarySensorEntityDescription,
+        entry: ConfigEntry,
+        system_zone_id: str,
+        zone_data: dict,
+    ) -> None:
+        """Initialize."""
+        super().__init__(coordinator, entry, system_zone_id, zone_data[AZD_NAME])
+        self._attr_name = f"{zone_data[AZD_NAME]} {description.name}"
+        self._attr_unique_id = f"{entry.entry_id}_{system_zone_id}_{description.key}"
+        self.entity_description = description
+
+    @property
+    def extra_state_attributes(self) -> Mapping[str, Any] | None:
+        """Return state attributes."""
+        _state_attr = None
+        if self.entity_description.key == AZD_DEMAND:
+            _state_attr = {
+                "air": self.get_zone_value(AZD_AIR_DEMAND),
+                "floor": self.get_zone_value(AZD_FLOOR_DEMAND),
```

Comment:
I suggest you add a subclass of BinarySensorEntityDescription where you add a list of extra attributes instead of this.

Reply:
I'm currently out of home, but I will do that as soon as possible.
Thanks for the suggestion!

Reply:
Changed in 45e69886536a855c7fb37976548fd2911c57cbf8

Reply:
Nice!